packages/zk-certificates/test/contracts/twitterVerificationProof.ts

```diff
@@ -0,0 +1,409 @@
+/* Copyright (C) 2023 Galactica Network. This file is part of zkKYC. zkKYC is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. zkKYC is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>. */
+import type { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import chai from 'chai';
+import hre, { ethers } from 'hardhat';
+import { groth16 } from 'snarkjs';
+
+import {
+  fromDecToHex,
+  fromHexToBytes32,
+  processProof,
+  processPublicSignals,
+} from '../../lib/helpers';
+import type { ZkCertificate } from '../../lib/zkCertificate';
+import {
+  generateSampleTwitterZkCertificate,
+  generateTwitterZkCertificateProofInput,
+} from '../../scripts/generateTwitterZkCertificateInput';
+import type { MockZkCertificateRegistry } from '../../typechain-types/contracts/mock/MockZkCertificateRegistry';
+import type { TwitterVerificationProof } from '../../typechain-types/contracts/TwitterVerificationProofProof';
+import type { TwitterVerificationProofVerifier } from '../../typechain-types/contracts/zkpVerifiers/TwitterVerificationProofVerifier';
+
+chai.config.includeStack = true;
+
+const { expect } = chai;
+
+describe('twitterVerificationProof SC', () => {
```

Comment:
For my taste, there is too much code duplication when comparing this test with the `twitterFollowersCountProof` test. This could be generalized, but it is not urgent and can be done later too.

Reply:
And the `twitterZkCertificate` test also looks the same.

Reply:
yeah, the differences are minor, in the tests that are relevant to verification status, follower thresholds, but let's leave it like this for now.